ehrapy/api/preprocessing/encoding/_encode.py

```diff
@@ -504,6 +594,99 @@ def _undo_encoding(
             layers={"original": new_x.copy()},
         )
 
+    @staticmethod
+    def _delete_all_encodings(adata: AnnData) -> Tuple[Optional[np.ndarray], Optional[list]]:
+        """Delete all encoded columns and keep track of their indices
+        Args:
+            adata: The AnnData object to operate on
+
+        Returns:
+            A temporary X were all encoded columns are deleted and all var_names of unencoded columns
+        """
+        var_names = list(adata.var_names)
+        if adata.X is not None and var_names is not None:
```

Comment:
Why is this check required? Looks like an assert to me?

Reply:
Ah y an `assert` may be sufficient here. Just wanted to cover some odd cases, but I even think this can be omitted entirely